infra/parse_signing_key.go

```diff
@@ -7,28 +7,44 @@ import (
 	"crypto/x509"
 	"encoding/pem"
 	"fmt"
+	"io"
+	"os"
 	"strings"
 
 	"github.com/cockroachdb/errors"
 
 	"github.com/checkmarble/marble-backend/utils"
 )
 
-func ParseOrGenerateSigningKey(ctx context.Context, privateKeyString string) *rsa.PrivateKey {
+func ReadParseOrGenerateSigningKey(ctx context.Context, privateKeyString string, keyFile string) *rsa.PrivateKey {
 	logger := utils.LoggerFromContext(ctx)
-	if privateKeyString == "" {
+	if privateKeyString == "" && keyFile == "" {
 		logger.InfoContext(ctx, "No AUTHENTICATION_JWT_SIGNING_KEY received, generating a new key pair. This is not recommended for production environments as the key will not be stable in time")
-		privateKey, err := generatePrivateKey(2048)
+		privateKey, err := generateRsaPrivateKey(2048)
 		if err != nil {
 			panic("Error generating RSA private key")
 		}
 		return privateKey
 	}
 
+	if keyFile != "" {
```

Comment:
this means that keyfile had precedence over privatekey, is it what we want ?

Reply:
Yea that's fine I think - anyway there is no reason at all to provide both, so I'm comfortable letting this at the customer's responsibility.